### PR TITLE
Check overflow when casting integer to decimal

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Part of #2986.

This deals with the case of casting integer to decimal.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by tustvold — APPROVED** [review]
I wonder if the two cast_integer_to_decimal methods could be combined together? :thinking:

**Comment by viirya** [comment]
> I wonder if the two cast_integer_to_decimal methods could be combined together? 🤔

It's doable. I combined them in latest commit.

**Comment by viirya** [comment]
Going to fix clippy errors: #3010

File: arrow-array/src/types.rs [hunk 1]
```diff
@@ -521,6 +524,10 @@ impl DecimalType for Decimal128Type {
         format_decimal_str(&value.to_string(), precision as usize, scale as usize)
     }
 
+    fn format_decimal_type(precision: u8, scale: u8) -> String {
```
Comment (tustvold) [inline arrow-array/src/types.rs:527]: The way we have handled this in other places is an associated const PREFIX

Comment (viirya) [reply arrow-array/src/types.rs]: Changed to `PREFIX` way.

**Pull request merged**